### PR TITLE
[Documentation] DICOM::Element library perldocification

### PR DESCRIPTION
**Pull request by cmadjar**
Using perldoc/perlpod to document `Element.pm` so that users can type in the terminal
`perldoc Element.pm` and get the documentation.

Using the `pod2markdown` tool from perl `Pod::Markdown` module, the `Element.md` file has been created so that we have a web displayed version of the documentation.
`pod2markdown Element.pm > Element.md`

Dependencies added: perldoc, Pod::Markdown, Pod::Usage

### CODE REVIEW

**Comment by MounaSafiHarab** [comment]
hey @cmadjar 

still needs work?

**Comment by cmadjar** [comment]
@MounaSafiHarab ready for review. This one is a bit hard to document for me so it might need more work later on for clarification...

File: dicom-archive/DICOM/Element.pm [hunk 1]
```diff
@@ -156,6 +237,27 @@ sub readFloat {
     return sprintf("%e", $val);
 }
 
+
+=pod
+
+=head3 readSequence($IN, $len)
+
+Reads Sequence.
+
```
Comment (nicolasbrossard) [inline dicom-archive/DICOM/Element.pm]: Skips over either a fixed number of bytes or over multiple sets of byte sequences delimited with specific byte values. When doing the latter, byte 0x00000000 is used to signal the end of the set of sequences. The sequence of bytes read is always discarded.

File: dicom-archive/DICOM/Element.pm [hunk 2]
```diff
@@ -156,6 +237,27 @@ sub readFloat {
...
+Three different cases:
+    - implicit Value Representation (VR), explicit length
+    - explicit VR, undefined length, items of defined length (w/end delimiter)
+    - implicit VR, undefined length, items of undefined length
+
+
+INPUTS:
+  - $IN : input file stream
+  - $len: total number of bytes in the field
```
Comment (nicolasbrossard) [inline dicom-archive/DICOM/Element.pm]: total number of bytes to skip, or 0 if all sequences should be skipped until the delimiter 0x00000000 is found.

File: dicom-archive/DICOM/Element.pm [hunk 3]
```diff
@@ -204,9 +306,19 @@ sub readSequence {
...
 
 
-# Return the Value Field length, and length before Value Field.
-# Implicit VR: Length is 4 byte int.
-# Explicit VR: 2 bytes hold VR, then 2 byte length.
+=pod
+
+=head3 readLength($IN)
+
+Reads length.
```
Comment (nicolasbrossard) [inline dicom-archive/DICOM/Element.pm]: Reads the length of a VR from a file, as an integer encoded on 16 or 32 bits.

File: dicom-archive/DICOM/Element.pm [hunk 4]
```diff
@@ -38,13 +53,38 @@ BEGIN {
...
+
+=head3 fill($IN, $dictref, $big_endian_image)
+
+Fills in C<self> from file.
+
+INPUTS:
+  - $IN              : input file
+  - $dictref         : DICOM dictionary
+  - $big_endian_image: if big endian image
+
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply dicom-archive/DICOM/Element.pm]: OK

File: dicom-archive/DICOM/Element.pm [hunk 5]
```diff
@@ -104,12 +144,24 @@ sub fill {
...
-#   bytelength: SHORT (2) or INT (4) bytes.
-#   fieldlength:Total number of bytes in the field.
-# If fieldlength > bytelength, multiple values are read in and
-# stored as a string representation of an array.
+
+=pod
+
+=head3 readInt($IN, $bytes, $len).
+
+Reads Int.
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply dicom-archive/DICOM/Element.pm]: 👍 thanks!

File: dicom-archive/DICOM/Element.pm [hunk 6]
```diff
@@ -104,12 +144,24 @@ sub fill {
...
+
+INPUTS:
+  - $IN   : input file stream.
+  - $bytes: SHORT (2) or INT (4) bytes.
+  - $len  : total number of bytes in the field.
+
+If C<fieldlength> > C<bytelength>, multiple values are read in and stored as a
+string representation of an array.
+
+RETURNS: string representation of an array
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply dicom-archive/DICOM/Element.pm]: 👍 thanks

File: dicom-archive/DICOM/Element.pm [hunk 7]
```diff
@@ -132,6 +184,19 @@ sub readInt {
   return $val;
 }
 
+
+=pod
+
+=head3 writeInt($OUT, $bytes)
+
+Writes Int into the output file C<$OUT>.
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply dicom-archive/DICOM/Element.pm]: 👍 done

File: dicom-archive/DICOM/Element.pm [hunk 8]
```diff
@@ -132,6 +184,19 @@ sub readInt {
...
+
+INPUTS:
+  - $OUT  : output file
+  - $bytes: number of bytes in the field
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply dicom-archive/DICOM/Element.pm]: done

File: dicom-archive/DICOM/Element.pm [hunk 9]
```diff
@@ -146,6 +211,22 @@ sub writeInt {
   }
 }
 
+
+=pod
+
+=head3 readFloat($IN, $format, $len)
+
+Reads Float.
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply dicom-archive/DICOM/Element.pm]: 👍 done. Thanks!

File: dicom-archive/DICOM/Element.pm [hunk 10]
```diff
@@ -146,6 +211,22 @@ sub writeInt {
...
+
+INPUTS:
+  - $IN    : input file stream
+  - $format: format of the variable
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply dicom-archive/DICOM/Element.pm]: done

File: dicom-archive/DICOM/Element.pm (hunk 1 above)
Comment (cmadjar) [reply dicom-archive/DICOM/Element.pm]: done

File: dicom-archive/DICOM/Element.pm (hunk 2 above)
Comment (cmadjar) [reply dicom-archive/DICOM/Element.pm]: done

File: dicom-archive/DICOM/Element.pm [hunk 11]
```diff
@@ -272,7 +384,16 @@ sub readLength {
   return ($vrstr, $length);
 }
 
-# Return the values of each field.
+
+=pod
+
+=head3 values()
+
+Returns the values of each field of the object.
```
Comment (nicolasbrossard) [inline dicom-archive/DICOM/Element.pm]: Returns the properties of a VR: group, element, offset, code, length, name and value.

File: dicom-archive/DICOM/Element.pm [hunk 12]
```diff
@@ -272,7 +384,16 @@ sub readLength {
...
+
+RETURNS: values of each field
```
Comment (nicolasbrossard) [inline dicom-archive/DICOM/Element.pm]: Returns the properties of a VR: group, element, offset, code, length, name and value.

File: dicom-archive/DICOM/Element.pm (hunk 3 above)
Comment (cmadjar) [reply dicom-archive/DICOM/Element.pm]: done

**Comment by cmadjar** [comment]
@nicolasbrossard Thanks! I integrated your comments :)

**Comment by MounaSafiHarab** [comment]
unassigned myself since Nick took charge :)

**Comment by nicolasbrossard** [comment]
Sorry...Didn't mean to be a thief....  ;)

**Comment by MounaSafiHarab** [comment]
can we change the base branch to be minor?

**Pull request merged**